tests/acceptance/features/webUISharingPublic/shareByPublicLink.feature

```diff
@@ -84,13 +85,33 @@ Feature: Share by public link
     Then the deleted elements should not be listed on the webUI
     And the deleted elements should not be listed on the webUI after a page reload
 
-  @skip @yetToImplement
-  Scenario: creating a public link with read permissions only makes it impossible to delete files via the link
-    When the user creates a new public link for folder "simple-folder" using the webUI with
-      | permission | read |
-    And the public accesses the last created public link using the webUI
+  Scenario: creating a public link with "Editor" role makes it possible to delete files via the link even with password set
+    Given user "user1" has shared folder "simple-folder" with link with "read, change, create, delete" permissions and password "pass123"
+    When the public uses the webUI to access the last public link created by user "user1" with password "pass123"
+    And the user deletes the following elements using the webUI
+      | name                                  |
+      | simple-empty-folder                   |
+      | lorem.txt                             |
+      | strängé filename (duplicate #2 &).txt |
+      | zzzz-must-be-last-file-in-folder.txt  |
+    Then the deleted elements should not be listed on the webUI
+
+  Scenario: creating a public link with "Viewer" role only makes it impossible to delete files via the link
+    Given user "user1" has shared folder "simple-folder" with link with "read" permissions
+    When the public uses the webUI to access the last public link created by user "user1"
     Then it should not be possible to delete file "lorem.txt" using the webUI
 
+  Scenario: creating a public link with "Editor" role makes it possible to upload files via the link even with password set
+    Given user "user1" has shared folder "simple-folder" with link with "read, change, create, delete" permissions and password "pass123"
+    When the public uses the webUI to access the last public link created by user "user1" with password "pass123"
+    And the user uploads file "new-lorem.txt" using the webUI
+    Then file "new-lorem.txt" should be listed on the webUI
```

Comment:
you could add 
```  
    And as "user1" file "new-lorem.txt" should exist
```

just to make sure the file really is available for the user and not only the UI lying

Reply:
uh oh
```
   ✔ Then file "new-lorem.txt" should be listed on the webUI # tests/acceptance/stepDefinitions/filesContext.js:214
   ✖ And as "user1" file "new-lorem.txt" should exist # tests/acceptance/stepDefinitions/webdavContext.js:8
       Error: file/folder should exist, but does not
```

I'll debug...

Reply:
Fixed. the path must be "simple-folder/new-lorem.txt" in that check